pandas/core/apply.py

```diff
@@ -274,13 +274,17 @@ def apply_standard(self):
 
         # we cannot reduce using non-numpy dtypes,
         # as demonstrated in gh-12244
-        if (
+        flag = (
             self.result_type in ["reduce", None]
             and not self.dtypes.apply(is_extension_array_dtype).any()
             # Disallow complex_internals since libreduction shortcut
             #  cannot handle MultiIndex
             and not isinstance(self.agg_axis, ABCMultiIndex)
-        ):
+            and len(set(self.dtypes))
+        )
+        return_result = None
```

Comment:
why is this needed?

Reply:
I'm using it to determine if line 314 was run.